src/functional/ffront/single_static_assign.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+import ast
+from dataclasses import dataclass, field
+
+
+class SingleStaticAssignPass(ast.NodeTransformer):
+    class RhsRenamer(ast.NodeTransformer):
+        """
+        Rename right hand side names.
+
+        Only read from parent visitor state, can not modify.
+        """
+
+        def __init__(self, state):
+            super().__init__()
+            self.state = SingleStaticAssignPass.State(name_counter=state.name_counter.copy())
+
+        def visit_Name(self, node: ast.Name) -> ast.Name:
+            if node.id in self.state.name_counter:
+                node.id = f"{node.id}_{self.state.name_counter[node.id]}"
```

Comment:
To avoid collisions with actual variable names, I would use "$" as separator (which is also common practice in compilers):
```suggestion
                node.id = f"{node.id}${self.state.name_counter[node.id]}"
```